compiler/packages/babel-plugin-react-compiler/src/__tests__/fixtures/compiler/preserve-memo-validation/spread-props.tsx

```diff
@@ -0,0 +1,31 @@
+import {useMemo} from 'react';
+
+function useSession() {
+  return {user: {userCode: 'ABC123'}};
+}
+
+function getDefaultFromValue(
+  defaultValues: string | undefined,
+  userCode: string,
+) {
+  return defaultValues ? `${defaultValues}-${userCode}` : userCode;
+}
+
+export function UpSertField({...props}: {defaultValues?: string}) {
+  const {
+    user: {userCode},
+  } = useSession();
+
+  const defaultValues = useMemo(
+    () => getDefaultFromValue(props.defaultValues, userCode),
+    [props.defaultValues, userCode],
+  );
+
+  return <div>{defaultValues}</div>;
```

Comment:
nit: we could probably simplify this example down, and then use `<ValidateMemoization />` (plus `sequentialRenders`) to test that its memoized as expected